File: rcs/riot-sdk/python/riot_sdk/core/order_state_page.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class OrderStateRecord:
    id: int | None
    order_id: str | None
    upper_id: str | None
    order_state: int | None
    appoint_vehicle_key: str | None
    execute_vehicle_key: str | None


@dataclass(frozen=True, slots=True)
class OrderStatePage:
    current: int | None
    size: int | None
    total: int | None
    records: tuple[OrderStateRecord, ...]

    @property
    def covers_all_records(self) -> bool:
        return (
            self.current == 1
            and self.total is not None
            and self.total >= 0
            and self.total == len(self.records)
            and (self.size is None or self.total <= self.size)
        )

File: rcs/riot-sdk/csharp/RIoT.Sdk.Core/OrderStatePage.cs
namespace RIoT.Sdk.Core;

/// <summary>
/// Stable order identity and assignment facts from the order-state page.
/// </summary>
public sealed record OrderStateRecord(
    long? Id,
    string? OrderId,
    string? UpperId,
    int? OrderState,
    string? AppointVehicleKey,
    string? ExecuteVehicleKey);

/// <summary>
/// Paged order facts with enough metadata for a caller to prove complete coverage.
/// </summary>
public sealed record OrderStatePage(
    long? Current,
    long? Size,
    long? Total,
    IReadOnlyList<OrderStateRecord> Records)
{
    public bool CoversAllRecords =>
        Current == 1 &&
        Total.HasValue &&
        Total.Value >= 0 &&
        Total.Value == Records.Count &&
        (!Size.HasValue || Total.Value <= Size.Value);
}
